arm64: Disable GENERIC_IRQ_EFFECTIVE_AFF_MASK

The effective affinity mask causes a lot of bugs by virtue of many
set_irq_affinity handlers only setting an effective affinity mask for an
IRQ's parent but not the IRQ itself. Since this is a widespread issue that
would require manual fixing on every different SoC, just disable the
effective affinity mask altogether and use the first CPU in an affinity
mask configured.

Signed-off-by: Sultan Alsawaf <sultan@kerneltoast.com>

--- a/drivers/irqchip/Kconfig	2024-04-02 11:32:10.895895646 +0800
+++ b/drivers/irqchip/Kconfig	2024-04-02 11:33:59.708208655 +0800
@@ -9,7 +9,6 @@
 	bool
 	depends on OF
 	select IRQ_DOMAIN_HIERARCHY
-	select GENERIC_IRQ_EFFECTIVE_AFF_MASK if SMP
 
 config ARM_GIC_PM
 	bool
@@ -35,7 +34,6 @@
 	bool
 	select IRQ_DOMAIN_HIERARCHY
 	select PARTITION_PERCPU
-	select GENERIC_IRQ_EFFECTIVE_AFF_MASK if SMP
 	select HAVE_ARM_SMCCC_DISCOVERY
 
 config ARM_GIC_V3_ITS

--- a/kernel/irq/Kconfig	2024-04-02 11:35:55.531621188 +0800
+++ b/kernel/irq/Kconfig	2024-04-02 11:36:34.844941030 +0800
@@ -24,8 +24,9 @@
 
 # Supports effective affinity mask
 config GENERIC_IRQ_EFFECTIVE_AFF_MASK
-       depends on SMP
        bool
+       depends on SMP
+       depends on !ARM64
 
 # Support for delayed migration from interrupt context
 config GENERIC_PENDING_IRQ

